### PR TITLE
Run Godep tool for build

### PR DESCRIPTION
**Pull request by dubeejw**
Closes https://github.com/apache/incubator-openwhisk-cli/issues/200

### CODE REVIEW

**Comment by dubeejw** [comment]
@houshengbo, please look at this. It seems that the godep tool is not running for the releases script.

File: .travis.yml [hunk 1]
```diff
@@ -48,6 +49,7 @@ before_deploy:
   - go get github.com/mattn/go-isatty
   - export build_file_name=wsk
   - export zip_file_name=OpenWhisk_CLI
+  - ./bin/godep restore -v
```
Comment (dubeejw) [inline .travis.yml]: For some reason two builds are happening. Why not just build once with travis.yml?????

**Comment by dubeejw** [comment]
https://github.com/apache/incubator-openwhisk-cli/pull/184 will supersede this. Just want a quick fix for now.

File: .travis.yml (hunk 1 above)
Comment (dubeejw) [reply .travis.yml]: We want to distribute binaries that are actually tested. Rebuild the binaries with the wrong dependencies (not using Godeps) and not testing those binaries is not good.

**Comment by houshengbo** [comment]
build: deps is used to build the binaries. It depends on deps, and deps runs "go get -d -t ./..." to install all the dependencies based on what is in the json file for deps. If the dependency is ready installed, there will be no change, no matter it is older or newer version. If the dependency is not available, it will install the version as json indicates.

One advantage I see in the Makefile is building up the dependencies for each task.

File: .travis.yml (hunk 1 above)
Comment (houshengbo) [reply .travis.yml]: No need to add this line here. As you see in test_openwhisk.sh, we run make build, so the dependecy deps runs as well. All the godep dependencies are installed by  "go get -d -t ./..." .

**Comment by dubeejw** [comment]
@houshengbo, we want to use the godep tool to install the CLI dependancies instead of `go get`. `godep` allows us to lock the dependency versions into place with ones that we know work and have approval to use legally.

File: Makefile [hunk 2]
```diff
@@ -9,13 +9,15 @@ BUILD=`git rev-parse HEAD`
 
 deps:
 	@echo "Installing dependencies"
-	go get -d -t ./...
+	./bin/godep restore -v
```
Comment (houshengbo) [inline Makefile]: This should be changed into godep restore -v

File: Makefile [hunk 3]
```diff
@@ -9,13 +9,15 @@ BUILD=`git rev-parse HEAD`
...
+	# go get -d -t ./...
```
Comment (houshengbo) [inline Makefile]: Remove this line

File: Makefile [hunk 4]
```diff
@@ -9,13 +9,15 @@ BUILD=`git rev-parse HEAD`
...
 
 LDFLAGS=-ldflags "-X main.CLI_BUILD_TIME=`date -u '+%Y-%m-%dT%H:%M:%S%:z'`"
 
-updatedeps:
-	@echo "Updating all dependencies"
-	@go get -d -u -f -fix -t ./...
+# Never want to do this ever
+# updatedeps:
+# @echo "Updating all dependencies"
+#	@go get -d -u -f -fix -t ./...
```
Comment (houshengbo) [inline Makefile]: Remove Line 17 to Line 20

**Comment by houshengbo** [comment]
@dubeejw I finally found out the issue. You are correct. See the comments.

**Comment by dubeejw** [comment]
Merged https://github.com/apache/incubator-openwhisk-cli/pull/202 instead.

File: .travis.yml (hunk 1 above)
Comment (mdeuser) [reply .travis.yml]: @dubeejw - good catch.  how many builds are now happening?

**Pull request closed without merging**